### DISCUSSION TITLE
Livewire: include blade file containing alpine data does not allow external x-data function.

### DISCUSSION THREAD

**Discussion by basilgass** [alpinejs/alpine#1026] [1. Help]
(Don't know how to correctly describe it in one single title)

I'm loading a laravel blade file using livewire. Basically, if a particular event happens, a blade page containing a div with x-data is loaded. This page contains, for example:

**resources/views/load-bug.blade.php**
```
<div class="bg-red-800 p-5 border rounded">
	<div x-data="loadfn()">
		<span x-text="foo"></span>
	</div>
	<script>
		function loadfn(){
			return {
				foo: 'bar'
			}
		}
		console.log(loadfn());
	</script>
</div>
```

Notice that the x-data is set with a function, instead of giving directly the object.
When the page is loaded, I receive this warning:
```alpine.js:115 Alpine Error: "ReferenceError: loadfn is not defined"```
and this error:
```Uncaught ReferenceError: loadfn is not defined```

But the console shows that the function exist !

Notice that if I change the x-data to include directly the object, eg:
```
<div class="bg-red-800 p-5 border rounded">
	<div x-data="{foo: 'bar'}">
		<span x-text="foo"></span>
	</div>
</div>
```
the page loads correctly and everything is fine.

The livewire component **livewire/load-bug.blade.php**
```
<div>
	<div wire:click="loadPage">The page to be loaded {{$page}}</div>
	@includeIf('themes.tools.' . $page)
</div>
```
The livewire controller **\app\Http\Livewire\LoadBug.php**
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class LoadBug extends Component
{
	public $page = 'hello world';
    public function render()
    {
        return view('livewire.load-bug');
    }

	public function loadPage(  ) {
		$this->page = 'load-bug';
    }
}
```

And add anywhere the code ```<livewire:load-bug />```

Is there a way to force alpine.js to find and use an external function ? Is it a known problem ? A bug ? Thanks for your help !

**Comment by ryangjchandler** [comment]
@basilgass This isn't a bug / problem with Alpine, it's more to do with where your JavaScript is being defined. You should probably be pushing the `<script>` tag to the end of the body with a `@stack` instead of putting it directly next to your component.

**Comment by basilgass** [comment]
Thanks for the quick reply :) But unfortunately, it doesn't work. I think the problem is that the dynamically added "section/page" is added afterward.

1. The page loads **completely**.
2. Click a button -> this will load/updated only a part using a blade file.
3. This file contains the main x-data container and the script tag. 

I don't think laravel can push this script with `@stack` without reloading the whole page ?

**Reply by KevinBatdorf** [reply]
Yeah, I don't think you can just inject a script tag like that. You need to instruct the browser to load it. There might be some libraries out there to help with that

**Reply by ryangjchandler** [reply]
Ah, okay. Yeah, `<script>` tags can't be injected at runtime with Livewire without use of a third-party library of some sorts. If you know that you're using that component on the page, can't you just include the JavaScript outside of the Blade file in the main page / layout. 

I know it ruins the organisation a little but it should work that way.
